### PR TITLE
GH-776, GH-775, GH-772 - fixing 404 handling, refactoring.

### PR DESCRIPTION
**Pull request by danielmkn**
Fixed 404 handling in following resources:
- resource/artifactory_group 
- resource/artifactory_permission_target 
- resource/artifactory_scoped_token
- resource/artifactory_ldap_setting_v2
- resource/artifactory_ldap_group_setting_v2
- resource/artifactory_*_user

Fixing GH-776, GH-775 and GH-772

### CODE REVIEW

File: pkg/artifactory/resource/security/resource_artifactory_group.go [hunk 1]
```diff
@@ -260,15 +260,14 @@ func (r *ArtifactoryGroupResource) Read(ctx context.Context, req resource.ReadRe
...
-		unableToRefreshResourceError(resp, err)
-		return
-	}
-
 	// Treat HTTP 404 Not Found status as a signal to recreate resource
 	// and return early
-	if response.StatusCode() == http.StatusBadRequest || response.StatusCode() == http.StatusNotFound {
-		resp.State.RemoveResource(ctx)
+	if err != nil {
+		if response.StatusCode() == http.StatusBadRequest || response.StatusCode() == http.StatusNotFound {
```
Comment (alexhung) [inline pkg/artifactory/resource/security/resource_artifactory_group.go:266]: We should only check for 404 for groups: https://jfrog.com/help/r/jfrog-rest-apis/get-group-details?tocId=iLr3VCaSAoke_By81l_HXw

File: pkg/artifactory/resource/security/resource_artifactory_scoped_token.go [hunk 2]
```diff
@@ -407,20 +407,18 @@ func (r *ScopedTokenResource) Read(ctx context.Context, req resource.ReadRequest
...
 	// and return early
-	if response.StatusCode() == http.StatusNotFound {
-		resp.Diagnostics.AddError(
-			fmt.Sprintf("Scoped token %s not found or not created", data.Id.ValueString()),
-			"Access Token would not be saved by Artifactory if 'expires_in' is less than the persistence threshold value (default to 10800 seconds) set in Access configuration. See https://www.jfrog.com/confluence/display/JFROG/Access+Tokens#AccessTokens-PersistencyThreshold for details."+err.Error(),
-		)
-		resp.State.RemoveResource(ctx)
-
+	if err != nil {
+		if response.StatusCode() == http.StatusBadRequest || response.StatusCode() == http.StatusNotFound {
```
Comment (alexhung) [inline pkg/artifactory/resource/security/resource_artifactory_scoped_token.go:413]: Same here. Only 404 is needed: https://jfrog.com/help/r/jfrog-rest-apis/get-token-by-id

File: pkg/artifactory/resource/user/resource_artifactory_anonymous_user.go [hunk 3]
```diff
@@ -82,22 +83,14 @@ func (r *ArtifactoryAnonymousUserResource) Read(ctx context.Context, req resourc
...
-		return
-	}
-
 	// Treat HTTP 404 Not Found status as a signal to recreate resource
 	// and return early
-	if response.StatusCode() == http.StatusNotFound {
-		resp.State.RemoveResource(ctx)
-
+	if err != nil {
+		if response.StatusCode() == http.StatusBadRequest || response.StatusCode() == http.StatusNotFound {
```
Comment (alexhung) [inline pkg/artifactory/resource/user/resource_artifactory_anonymous_user.go:89]: 404 only: https://jfrog.com/help/r/jfrog-rest-apis/get-user-details?tocId=s80tAWGpQA2~An1Ardf9Yw

**Review by alexhung — CHANGES_REQUESTED** [review]
Some of the resources we should only check for 404

File: pkg/artifactory/resource/security/resource_artifactory_group.go (hunk 1 above)
Comment (danielmkn) [reply pkg/artifactory/resource/security/resource_artifactory_group.go:266]: Removed.

File: pkg/artifactory/resource/security/resource_artifactory_scoped_token.go (hunk 2 above)
Comment (danielmkn) [reply pkg/artifactory/resource/security/resource_artifactory_scoped_token.go:413]: Removed.

File: pkg/artifactory/resource/user/resource_artifactory_anonymous_user.go (hunk 3 above)
Comment (danielmkn) [reply pkg/artifactory/resource/user/resource_artifactory_anonymous_user.go:89]: Removed.

**Review by alexhung — DISMISSED** [review]

**Review by alexhung — APPROVED** [review]

**Pull request merged**